rust-runtime/aws-smithy-http/src/operation/error.rs

```diff
@@ -185,3 +186,21 @@ impl Error for BuildError {
         }
     }
 }
+
+impl From<BuildError> for DeserializeError {
+    fn from(build_error: BuildError) -> Self {
+        DeserializeError::custom_source(
```

Comment:
This conversion should probably live in the generated crate.

Otherwise with this  _all_ clients (that have fallible builders and hence use `BuildError` from `aws-smithy-http`) will depend on `aws-smithy-json` for this conversion, _regardless of their protocol_. So e.g. restXml clients will start depending on the `aws-smithy-json` crate, compiling a lot of unused code and causing confusion. See https://github.com/awslabs/smithy-rs/issues/1947, which is related.